internal/providers/pluginfw/products/permissions/resource_permission.go

```diff
@@ -0,0 +1,652 @@
+	mapping, err := permissions.GetResourcePermissionsFromId(objectID)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get resource permissions mapping", err.Error())
+		return
+	}
+
+	// Read current permissions
+	idParts := strings.Split(objectID, "/")
+	permID := idParts[len(idParts)-1]
+
+	perms, err := w.Permissions.Get(ctx, iam.GetPermissionRequest{
+		RequestObjectId:   permID,
+		RequestObjectType: mapping.GetRequestObjectType(),
+	})
+	if err != nil {
+		// If the object or permissions are not found, remove from state to trigger recreation
+		if apierr.IsMissing(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Failed to read permissions", err.Error())
+		return
+	}
+
+	// Filter for the specific principal
+	found := false
+	var currentPermissionLevel types.String
+	for _, acl := range perms.AccessControlList {
+		if r.matchesPrincipal(acl, principal) {
+			// Update the state with the current permission level
+			if len(acl.AllPermissions) > 0 {
+				currentPermissionLevel = types.StringValue(string(acl.AllPermissions[0].PermissionLevel))
+				found = true
+				break
+			}
+		}
+	}
+
+	if !found {
+		// Permission for this specific principal no longer exists, remove from state to trigger recreation
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// Read the object identifier field from current state to preserve it
+	// (It should already be in state, but we need to make sure it stays there)
+	var objectFieldValue types.String
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root(mapping.GetField()), &objectFieldValue)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update state using SetAttribute
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(mapping.GetField()), objectFieldValue)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("permission_level"), currentPermissionLevel)...)
+}
+
+func (r *PermissionResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Read ID, principals, and permission_level from plan using GetAttribute
+	var id, userName, groupName, servicePrincipalName, permissionLevel types.String
+
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("id"), &id)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("user_name"), &userName)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("group_name"), &groupName)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("service_principal_name"), &servicePrincipalName)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("permission_level"), &permissionLevel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	w, err := r.Client.WorkspaceClient()
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get workspace client", err.Error())
+		return
+	}
+
+	// Parse the ID to get object ID and principal
+	objectID, _, err := r.parseID(id.ValueString())
```

Comment:
This should not be necessary in update, since the object ID should be in the PermissionResourceModel